fix(migrations): add protractor support if protractor imports are detected

crisbeto opened the pull request:
The new `bootstrapApplication` API doesn't include Protractor support anymore which may cause existing e2e tests to break after the migration. These changes add some logic that will provide Protractor support if any imports to `protractor` or `protractor/*` are detected.

devversion commented on packages/core/schematics/ng-generate/standalone-migration/standalone-bootstrap.ts:
```diff
@@ -703,3 +720,22 @@ function getLastImportEnd(sourceFile: ts.SourceFile): number {
+
+/** Checks if any of the program's files has an import of a specific module. */
+function hasImport(program: NgtscProgram, moduleName: string): boolean {
+  // Check all the files in the program since tests might've been excluded.
```
Is that really necessary? I would expect the parsed tsconfig rootnames to include all the relevant spec files. I can't think of cases where spec files would be transitively brought into the program due to something importing them.

(also this would speed it up a little, as the program could become quite large in g3)

crisbeto replied:
The comment is referring to the fact that this function is fetching all the files through `getSourceFiles` instead of using the filtered files that are passed in to `toStandaloneBootstrap`.

devversion replied:
yeah, but why do you need `getSourceFiles`, if `[rootNames`](https://github.com/angular/angular/blob/main/packages/core/schematics/utils/typescript/compiler_host.ts#L54) would potentially more appropriate?

crisbeto replied:
`getSourceFiles` returns an array of the `SourceFile` ASTs whereas with `rootNames` one more lookup would have to happen.

crisbeto replied:
Ah, now I see your point about g3. `getSourceFiles` would also include 3p files.

devversion reviewed (approved):
LGTM, just one optional comment.

crisbeto committed:
fix(migrations): add protractor support if protractor imports are detected

The new `bootstrapApplication` API doesn't include Protractor support anymore which may cause existing e2e tests to break after the migration. These changes add some logic that will provide Protractor support if any imports to `protractor` or `protractor/*` are detected.
